build.sbt

```diff
@@ -43,11 +43,22 @@ ThisBuild / libraryDependencies ++= Seq(
     TestDeps.scalatestplusScalacheck,
 )
 
+////////////////////////////
+// Linting and formatting //
+////////////////////////////
+
+// scalafmt
 // TODO: Remove if we decide we are happy with allways reformatting all
 // Only check/fix against (tracked) files that have changed relative to the trunk
 // ThisBuild / scalafmtFilter := "diff-ref=origin/unstable"
 ThisBuild / scalafmtPrintDiff := true
 
+// scalafix
+// https://scalacenter.github.io/scalafix/docs/rules/RemoveUnused.html
+ThisBuild / scalacOptions ++= Seq("-Ywarn-unused")
+ThisBuild / semanticdbEnabled := true
```

Comment:
I see scalafix is wired through a specific make target, so you might want to use [`scalafixEnable`](https://scalacenter.github.io/scalafix/docs/users/installation.html#enable-semanticdb-for-current-shell-session) as a first step of that command to avoid the overhead of semanticdb for everyone upon compilation.